plugins/reolink/src/main.ts

```diff
@@ -318,10 +324,28 @@ class ReolinkCamera extends RtspSmartCamera implements Camera, DeviceProvider, R
         }
         if (this.hasSiren())
             interfaces.push(ScryptedInterface.DeviceProvider);
+        if (this.hasBattery()) {
+            interfaces.push(ScryptedInterface.Battery, ScryptedInterface.Online);
+            await this.startBatteryCheckInterval();
+        }
 
         await this.provider.updateDevice(this.nativeId, name, interfaces, type);
     }
 
+    async startBatteryCheckInterval() {
+        if (this.batteryTimeout) {
+            clearInterval(this.batteryTimeout);
+        }
+
+        this.batteryTimeout = setInterval(async () => {
+            const api = this.getClientWithToken();
+
+            const { batteryPercent, sleep } = await api.getBatteryInfo();
+            this.batteryLevel = batteryPercent;
+            this.online = !sleep;
```

Comment:
This sleep parameter is a very useful thing to keep to build more stuff later on, not sure if it's the right place where to store it? It tells if the camera is in standby. An use case I use it for is to generate a snapshot as soon as it comes online to save battery even more